src/pbft/pbft.h

```diff
@@ -181,13 +185,36 @@ namespace pbft
         std::make_unique<ClientProxy<kv::TxHistory::RequestID, void>>(
           *message_receiver_base);
 
-      auto cb = [](Reply* m, void* ctx) {
+      auto reply_handler_cb = [](Reply* m, void* ctx) {
         auto cp =
           static_cast<ClientProxy<kv::TxHistory::RequestID, void>*>(ctx);
         cp->recv_reply(m);
       };
+      message_receiver_base->register_reply_handler(
+        reply_handler_cb, client_proxy.get());
 
-      message_receiver_base->register_reply_handler(cb, client_proxy.get());
+      auto batch_committed_cb =
+        [](uint64_t batch_id, void* batch, uint32_t size, void* ctx) {
+          auto ledger = static_cast<raft::LedgerEnclave*>(ctx);
+
+          struct LedgerEntry
```

Comment:
Could this structure be defined somewhere else (e.g. a new `pbftledger.h` file?).